### PR TITLE
Install Go versions alongside each other

### PR DESCRIPTION
**Pull request by mikemcquaid**
- Add suffixes to Go `bin` commands
- Add `go14` formula
- Make some syntax changes to better match the main `go` formula

CC @DomT4 @jhosh

### CODE REVIEW

File: go14.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  version "1.4.2"
+  sha256 "299a6fd8f8adfdce15bc06bde926e7b252ae8e24dd5b16b7d8791ed79e7b5e9b"
+
+  option "with-cc-all", "Build with cross-compilers and runtime support for all supported platforms"
+  option "with-cc-common", "Build with cross-compilers and runtime support for darwin, linux and windows"
+  option "without-cgo", "Build without cgo"
+  option "without-godoc", "godoc will not be installed for you"
+  option "without-vet", "vet will not be installed for you"
+
+  deprecated_option "cross-compile-all" => "with-cc-all"
```
Comment (DomT4) [inline go14.rb]: There are no `deprecated_option`s in new formulae :trollface:

Comment (mikemcquaid) [inline go14.rb]: Oops, too late.

Comment (mikemcquaid) [inline go14.rb]: 7978cca

File: go13.rb [hunk 2]
```diff
@@ -65,10 +67,11 @@ def install
     (buildpath/"pkg/obj").rmtree
 
     libexec.install Dir["*"]
-    bin.install_symlink Dir["#{libexec}/bin/go*"]
+    (bin/"go13").write_env_script(libexec/"bin/go", :PATH => "#{libexec}/bin:$PATH")
```
Comment (josh) [inline go13.rb]: Nice, love the version binstub.

**Pull request closed without merging**